app/src/main/java/org/p2p/wallet/settings/ui/settings/SettingsPresenter.kt

```diff
@@ -39,9 +40,14 @@ class SettingsPresenter(
 
     override fun logout() {
         launch {
-            authLogoutInteractor.onUserLogout()
-            adminAnalytics.logSignedOut()
-            appRestarter.restartApp()
+            try {
+                authLogoutInteractor.onUserLogout()
```

Comment:
you remove suspend modifier from onUserLogout() fun, so maybe we will remove scope or launch runBlocking to make sure that we remove everything before restart app